pkg/cloudprovider/providers/oci/load_balancer_security_lists.go

```diff
@@ -408,22 +408,15 @@ func getNodeIngressRules(
 			ingressRules = append(ingressRules, rule)
 			desiredHealthChecker.Delete(*rule.Source)
 			continue
-		}
-
-		inUse, err := healthCheckPortInUse(serviceLister, int32(desiredPorts.HealthCheckerPort))
-		if err != nil {
-			// Unable to determine if this port is in use by another service, so I guess
-			// we better err on the safe side and keep the rule.
-			logger.Errorf("failed to determine if port: %d is still in use: %v", desiredPorts.HealthCheckerPort, err)
-			ingressRules = append(ingressRules, rule)
-			continue
-		}
-		if inUse {
-			// This rule is no longer needed for this service, but is still used
-			// by another service, so we must still keep it.
-			logger.Infof("Port %d still in use by another service.", desiredPorts.HealthCheckerPort)
-			ingressRules = append(ingressRules, rule)
-			continue
+		} else if *r.Max == desiredPorts.HealthCheckerPort {
+			inUse, err := healthCheckPortInUse(serviceLister, int32(desiredPorts.HealthCheckerPort))
+			if err != nil {
+				logger.Errorf("failed to determine if port: %d is still in use: %v", desiredPorts.HealthCheckerPort, err)
+				ingressRules = append(ingressRules, rule)
```

Comment:
Should these still have `continue` to avoid future problems if additional checks are added outside the if/else?

Reply:
At the moment, only because of the debug log line at 424 that states that the rule is no longer in use, and will be deleted.

But to your point about the ```continue``` statements - these all tell me that there are probably one to many functions in here that could be lifted.